OpenRA.Mods.Common/Widgets/Logic/ServerCreationLogic.cs

```diff
@@ -116,20 +116,20 @@ public ServerCreationLogic(Widget widget, ModData modData, Action onExit, Action
 			if (noticesNoUPnP != null)
 			{
 				noticesNoUPnP.IsVisible = () => advertiseOnline &&
-					(UPnP.Status == UPnPStatus.NotSupported || UPnP.Status == UPnPStatus.Disabled);
+					(Game.UPnP.Status == UPnPStatus.NotSupported || Game.UPnP.Status == UPnPStatus.Disabled);
```

Comment:
This is probably missing a null check.

Reply:
Relying on all consumers to null-check `Game.UPnP` really isn't great, and is asking for crashes.

How many things inside `UPnP` actually need to be publicly called? It might be better to expose a `Game.UPnPStatus` property that internally does a null check (and returns NotSupported), and similarly expose the other UPnP methods as wrappers on Game if its only the two that create and remove mappings.

Reply:
I made sure that the null check is not required anymore. Tested all 3 setups: no network, router, NAT discovery turned on. Also renamed things to reflect that Apple's NAT-PMP is supported as well.

Reply:
Did you test the dedicated server?

Reply:
Forgot, but did now. It does not have NAT support though.